docs/source/exposing.md

```diff
@@ -0,0 +1,267 @@
+Example:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    nodeService:
+     type: LoadBalancer
+     loadBalancerClass: my-custom-load-balancer-class
+```
+
+---
+
+### Broadcast Options
+
+Broadcast options controls what is the source of the address being broadcasted to other clients and nodes.
+It's configured independently for clients and nodes because you may want to expose these two types of traffic on different networks, due to costs or security concerns.
+
+#### `PodIP` Type
+
+Address broadcasted to clients/nodes is taken from Pod.
+Because Pod address may be ambiguous, you may provide what's the source by setting `podIP.source`.
+By default, address is taken from Pod `status.PodIP`.
+
+Example:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    broadcastOptions:
+       clients:
+         type: PodIP
+         podIP:
+           source: Status
+```
+
+#### `ServiceClusterIP` Type
+
+Address broadcasted to clients/nodes is taken from node dedicated Service, from `spec.ClusterIP` field.
+
+In order to configure it, your `nodeService` template must deploy Service having ClusterIP.
+
+Example:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    broadcastOptions:
+       clients:
+         type: ServiceClusterIP
+```
+
+#### `ServiceLoadBalancerIngress` Type
+
+Address broadcasted to clients/nodes is taken from the node dedicated Service, from `status.ingress[0].ipAddress` or `status.ingress[0].hostname` field.
+
+In order to configure it, your `nodeService` template must deploy LoadBalancer Service.
+
+Example:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    broadcastOptions:
+       clients:
+         type: ServiceLoadBalancerIngress
+         podIP:
+           source: Status
+```
+
+## Deployment Examples
+
+In this section, we will showcase specific examples of various network scenarios and explain how nodes and clients communicate with one another.
+### In-cluster only
+
+ScyllaCluster definition:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    nodeService:
+      type: ClusterIP
+    broadcastOptions:
+      clients:
+        type: ServiceClusterIP
+      nodes:
+        type: ServiceClusterIP
+```
+
+Both client and nodes are deployed within the same Kubernetes cluster. 
+They talk through IP addresses from ClusterIP Service.
+Because ClusterIP Services are only routable within the same Kubernetes cluster, connection from outside is impossible.
+
+![ClusterIPs](static/exposing/clusterip.svg)
+
+### In-cluster node-to-node, VPC clients-to-nodes
+
+ScyllaCluster definition:
+```yaml
+apiVersion: scylla.scylladb.com/v1
+kind: ScyllaCluster
+spec:
+  exposeOptions:
+    nodeService:
+      type: ClusterIP
+    broadcastOptions:
+      clients:
+        type: PodIP
+      nodes:
+        type: ServiceClusterIP
```

Comment:
Any reason why not having it? It's something user can configure, no matter what's the reason behind it. This sections showcases how traffic flows to let users better understand the outcome of every option, not to recommend how they should deploy their application.

Reply:
I just think it's just somewhat confusing when compared to the next example which only uses PodIPs - having examples like these is usually meant to showcase the recommended configuration for different setups, so this sort of implies that there's something different about this scenario which calls for using ClusterIPs.

Reply:
Would you mind elaborating what is confusing about this example compared to the next one?

> having examples like these is usually meant to showcase the recommended configuration

Example is not recommendation. I haven't used "recommended" word anywhere in this document. 

>  this sort of implies that there's something different about this scenario which calls for using ClusterIPs.

isn't it a different scenario?

Reply:
> Would you mind elaborating what is confusing about this example compared to the next one?

The description doesn't say anything about why you'd want to use CluserIPs for node to node communication. Which I think is confusing, especially since the first sentence in the description is "In this scenario, we assume that the Pod IP subnet is routable within a VPC.".

> Example is not recommendation. I haven't used "recommended" word anywhere in this document.

Well, given there are no other recommendations for the network setup in out docs, users are going to use them as such.

> isn't it a different scenario?

It is, but my point is there isn't anything specific about it that would explain the different setup, other than that I can *also* be set up this way.

Reply:
> The description doesn't say anything about why you'd want to use CluserIPs for node to node communication. 

Why would I describe something that I don't intent to explain? I don't need argument behind this scenario to show how traffic flows when cluster is configured this way.

> especially since the first sentence in the description is "In this scenario, we assume that the Pod IP subnet is routable within a VPC.".

Isn't it a required assumption for client-node-traffic available in this scenario?

>  my point is there isn't anything specific about it that would explain the different setup, other than that it can also be set up this way.


So you suggest to remove it? How not having anything would help understanding flows in different scenarios?

Reply:
I understand your point about this only showcasing the traffic flow, but I'm pretty sure people are going to use it as a reference for how to set up their scyllaclusters in their particular network setups, that's why I think it could use some explanation. Anyway I now see that it's explained in the earlier sections, so I'm resolving this comment.